hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -82,6 +85,44 @@ public class DatanodeAdminMonitorImpl implements DatanodeAdminMonitor {
   private long unhealthyContainers = 0;
   private long underReplicatedContainers = 0;
 
+  @SuppressFBWarnings(value = "SIC_INNER_SHOULD_BE_STATIC")
+  private final class ContainerStateInWorkflow {
```

Comment:
Rather than suppress the FB warning, can this class be made `private final static class ...`?  I am not an expert in this area, but usually inner classes I've seen are static. The difference between static and non static inner classes seems to be that "non static" inner classes can directly access the enclosing classes instance variables and methods.

A static inner class cannot directly access the enclosing classes methods. It has to do it via an object reference.

In this case, the inner class is a simple wrapper around a set of variables and does not need to access the enclosing methods class, and therefore can be static I think.

Reply:
Thanks.  Removed the suppress annotation and properly converted _instead_ to` static final nested class` _from_ the` final inner class.`  As the inner class does not refer to the outer class instance, it should indeed be a static nested class.